backend/uclapi/libcal/serializers.py

```diff
@@ -0,0 +1,253 @@
+        min_value=0,
+        required=False,
+        help_text='Pass a location id here to only show bookings for that location.'
+    )
+    date = serializers.DateField(
+        required=False,
+        help_text='The date to retrieve bookings. Dates in the past are ignored.'
+    )
+    days = serializers.IntegerField(
+        min_value=0,
+        max_value=365,
+        required=False,
+        help_text='The number of days into the future to retrieve bookings from, starting from [date] parameter.'
+    )
+    limit = serializers.IntegerField(
+        min_value=1,
+        max_value=500,
+        required=False,
+        help_text='How many bookings to return, max value is 500.'
+    )
+
+
+class LibCalPersonalBookingsGETSerializer(LibCalBookingsGETSerializer):
+    email = serializers.EmailField(
+        required=True,  # Default, but stated for clarity.
+        help_text='Pass an email address to only show bookings made by that patron.'
+    )
+    formAnswers = serializers.IntegerField(
+        min_value=0,
+        max_value=1,
+        required=False,
+        help_text=(
+            'Flag to indicate if you want custom form answers to be returned. You can hit the /equipment/question'
+            'endpoint to retrieve the details of the booking form questions.'
+        )
+    )
+
+
+class BookingSerializer(serializers.Serializer):
+    id = serializers.IntegerField(
+        min_value=0,
+        required=True,
+        help_text='Pass a location id here to only show bookings for that location.'
+    )
+    seat_id = serializers.IntegerField(
+        min_value=0,
+        required=False,
+        help_text='Pass a location id here to only show bookings for that location.'
+    )
+    to = serializers.DateTimeField(
+        required=True,
+        help_text='')
+
+
+class LibCalReservationPOSTSerializer(serializers.Serializer):
+    start = serializers.DateTimeField(
+        required=True,
+        help_text='Booking start date/time in ISO8601 format.'
+    )
+    fname = serializers.CharField(
+        max_length=100,
+        required=True,
+        help_text='First name of person making the booking.'
+    )
+    lname = serializers.CharField(
+        max_length=100,
+        required=True,
+        help_text='Last name of person making the booking.'
+    )
+    test = serializers.IntegerField(
+        min_value=0,
+        max_value=1,
+        help_text=(
+            'A flag to indicate if this is a test booking. If this flag is set the system will process the booking'
+            'rules but not actually make the booking. This is a useful feature when developing an application that'
+            'makes use of the booking API.'
+        )
+    )
+    email = serializers.EmailField(required=True, help_text='Email address of person making the booking.')
+    nickname = serializers.CharField(max_length=100, required=False)
+    # TODO: custom form question ids????? See LibCal API Docs.
```

Comment:
I think we must include this before merging, this is required for the UCL NHS work spaces.

Reply:
Yeah it's the NHS ones that have questions. Now I just need to figure out the name of the field...

Reply:
Looks like they're not asking for the nhs number anymore